src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -112,7 +117,10 @@ internal partial class RequestExecutionQueue
         public void Shutdown()
         {
             _cancelSource.Cancel();
-            DrainQueue();
+
+            // Tell the queue not to accept any more items.
+            _queue.Complete();
```

Comment:
consider commenting that this will allow it to just bail, and that the GC will clean up everything.  It also means we won't complete requests made to the server, but that is an acceptable thing for servers to do and clients should expect that with shutting down servers.

--

aside, *is* there an appropriate "i'm shutting down" message that servers are supposed to (or can) givein response to messages to help clients out?